File: app/src/main/java/com/example/shibegallery/repository/remote/ShibeService.java
package com.example.shibegallery.repository.remote;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ShibeService {

    @GET("{animal}")
    Call<List<String>> getImageUrls(
            @Path("animal") String path,
            @Query("count") int count
    );
}

File: appkotlin/src/main/java/com/example/appkotlin/repository/remote/ShibeService.kt
package com.example.appkotlin.repository.remote

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ShibeService {

    @GET("{animal}")
    fun getAnimal(
            @Path("animal") animal: String,
            @Query("count") count: Int
    ) : Call<List<String>>
}